app/models/company.rb

```diff
@@ -1,6 +1,20 @@
 class Company < ActiveRecord::Base
+  include ActiveModel::Dirty
+
   validates :name, :organization, :title, :state, :city, presence: true
   validates :email, presence: true, length: { maximum: 255 },
                     format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i, }
 
+  geocoded_by :full_address
+
+  after_validation :geocode, if: ->(obj){(obj.city_changed? || obj.state_changed?)}
```

Comment:
Basically, I only wanted to geocode if the city or state changed. This keeps our api queries low.